client/cmd/resource/change_namespace.go

```diff
@@ -0,0 +1,140 @@
+	}
+	cmd := &cobra.Command{
+		Use:      "change-namespace",
+		Short:    "Create namespace of a resource",
+		Example:  "optimus resource change-namespace <resource-name> --old-namespace <old-namespace> --new-namespace <new-namespace>",
+		Args:     cobra.MinimumNArgs(1),
+		PreRunE:  changeNamespace.PreRunE,
+		RunE:     changeNamespace.RunE,
+		PostRunE: changeNamespace.PostRunE,
+	}
+	// Config filepath flag
+	cmd.Flags().StringVarP(&changeNamespace.configFilePath, "config", "c", config.EmptyPath, "File path for client configuration")
+	internal.MarkFlagsRequired(cmd, []string{"old-namespace", "new-namespace"})
+	changeNamespace.injectFlags(cmd)
+
+	return cmd
+}
+
+func (c *changeNamespaceCommand) injectFlags(cmd *cobra.Command) {
+	// Mandatory flags
+	cmd.Flags().StringVarP(&c.oldNamespace, "old-namespace", "o", "", "current namespace of the resource")
+	cmd.Flags().StringVarP(&c.newNamespace, "new-namespace", "n", "", "namespace to which the resource needs to be moved to")
+
+	// Mandatory flags if config is not set
+	cmd.Flags().StringVarP(&c.project, "project-name", "p", "", "Name of the optimus project")
+	cmd.Flags().StringVar(&c.host, "host", "", "Optimus service endpoint url")
+}
+
+func (c *changeNamespaceCommand) PreRunE(_ *cobra.Command, _ []string) error {
+	// Load mandatory config
+	conf, err := config.LoadClientConfig(c.configFilePath)
+	if err != nil {
+		return err
+	}
+
+	c.clientConfig = conf
+	return err
+}
+
+func (c *changeNamespaceCommand) RunE(_ *cobra.Command, args []string) error {
+	resourceName := args[0]
+	err := c.sendChangeNamespaceRequest(resourceName)
+	if err != nil {
+		return fmt.Errorf("namespace change request failed for resource %s: %w", resourceName, err)
+	}
+	c.logger.Info("successfully changed namespace and deployed new DAG on Scheduler")
+	return nil
+}
+
+func (c *changeNamespaceCommand) sendChangeNamespaceRequest(resourceName string) error {
+	conn, err := connectivity.NewConnectivity(c.host, changeNamespaceTimeout)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	// fetch Instance by calling the optimus API
+	resourceRunServiceClient := pb.NewResourceServiceClient(conn.GetConnection())
+	request := &pb.ChangeResourceNamespaceRequest{
+		ProjectName:      c.project,
+		NamespaceName:    c.oldNamespace,
+		DatastoreName:    c.dataStore,
+		ResourceName:     resourceName,
+		NewNamespaceName: c.newNamespace,
+	}
+
+	_, err = resourceRunServiceClient.ChangeResourceNamespace(conn.GetContext(), request)
+	return err
+}
+
+func (c *changeNamespaceCommand) PostRunE(_ *cobra.Command, args []string) error {
+	c.logger.Info("\n[INFO] Moving resource in filesystem")
+	resourceName := args[0]
+	fs := afero.NewOsFs()
+	var source, destination string
+	for _, namespace := range c.clientConfig.Namespaces {
+		if namespace.Name == c.oldNamespace {
+			for _, datastore := range namespace.Datastore {
+				if datastore.Type == c.dataStore {
+					source = "./" + datastore.Path + "/" + resourceName
```

Comment:
I don't think we need to add `./` in this line (and below line for destination). Also, for path, please use `path.Join()` instead.